ARCs/arc-0060.md

```diff
@@ -0,0 +1,449 @@
+    - the wallet **MUST** reject if `data.bytes` field is prepended with a forbidden domain separator `TX` or `TG`.
+- `signer`:
+  - it **MUST** be a valid `Ed25519Pk` object, otherwise the wallet **MUST** reject.
+  - the wallet **MUST** reject if the `signer` is unknown.
+  - if `hdPath` is specified, the wallet **MUST** reject if `signer` does not correspond the public key derived with the `hdPath` parameters.
+
+- `hdPath`:
+  - The wallet **MAY** not support this field. In that case, it **MUST** throw a `4200` error.
+  - if specified, it **MUST** be a valid `HDWalletMetadata` object, otherwise the wallet **MUST** reject.
+  - if specified and supported, the wallet **MUST** verify that the derivation path resolves to a public key corresponding to `signer`.
+
+#### Semantic of `StdSignMetadata`
+
+- `scope`:
+  - it **MUST** be a valid `ScopeType` string, otherwise the wallet **MUST** throw a `4600` error.
+
+- `schema`:
+  - it **MUST** be a JSON schema with the properties `ARC60Domain` and `bytes`, otherwise the wallet **MUST** throw a `4601` error.
+
+- `encoding`:
+  - if specified, it **MUST** be used to encode `data.bytes`.
+
+#### General Validation
+
+Every input of the `signData(signingData, metadata)` must be validated.
+
+The validation:
+
+- **SHALL NOT** rely on TypeScript typing as this can be bypassed. Types **MUST** be manually verified.
+- **SHALL NOT** assume that `signingData.data` complies with `metadata.schema`. The schema **MUST** be manually verified and all the required parameters **MUST** be checked.
+- **SHALL NOT** assume to compute signatures with the Algorand SDK `signBytes` function. There is no indirect validation of the passed `signingData` and `metadata` objects on that function. In general, the validation **SHALL NOT** rely on underlying SDKs to validate the inputs of an ARC-60 signing operation.
+
+#### Display message
+
+The wallet **MUST** display a message for any signing request. The message **MUST** be computed dynamically taking as inputs the `metadata.scope`, the `signingData.signer`, and the message being signed `msg`.
+
+> The `message` structure is left to the implementation. An example is:
+>
+> `You are about to sign <msg> with scope <metadata.scope> and signer <signingData.signer>`.
+
+The wallet **SHOULD** display a warning in the message if the `ScopeType` is a `LSIG`.
+
+## Rationale
+
+This API was designed to enable the secure signing of structured data with Algorand wallets. The API:
+
+- Is generic to _pure_ ed25519 key pairs and does not require key manipulation
+- Is easy to extend with custom JSON schema to support future signing use cases (e.g. authentication)
+- Is secure by design preventing malicious applications from tricking users into signing malicious data or in the worst cases approving malicious transactions.
+
+This API was not designed to sign Algorand transactions or a group of transactions (see ARC-1 instead).
+
+## Backwards Compatibility
```

Comment:
I'm wondering if we should even mention ARC-47 in this ARC here. I'm inclined to remove it for the sake of simplicity and to keep this ARC as short and concise as possible. If there's demand for signing lsigs with ARC-60, somebody can propose a new ARC, specify the scope, and define the schema, etc.